main.tf

```diff
@@ -0,0 +1,99 @@
+module "label" {
+  source      = "git::https://github.com/cloudposse/terraform-null-label.git?ref=master"
+  environment = "${var.environment}"
+  name        = "${var.name}"
+  namespace   = "${var.namespace}"
+}
+
+data "template_file" "cloud_init_cloudwatch_agent" {
+  template = "${file("${path.module}/templates/cloud_init.yaml")}"
+
+  vars {
+    cloudwatch_agent_configuration = "${var.metrics_config == "standard" ? base64encode(data.template_file.cloudwatch_agent_configuration_standard.rendered) : base64encode(data.template_file.cloudwatch_agent_configuration_advanced.rendered)}"
+  }
+}
+
+data "template_file" "cloudwatch_agent_configuration_advanced" {
+  template = "${file("${path.module}/templates/cloudwatch_agent_configuration_advanced.json")}"
+
+  vars {
+    aggregation_dimensions      = "${jsonencode(var.aggregation_dimensions)}"
+    cpu_resources               = "${var.cpu_resources}"
+    disk_resources              = "${jsonencode(var.disk_resources)}"
+    metrics_collection_interval = "${var.metrics_collection_interval}"
+  }
+}
+
+data "template_file" "cloudwatch_agent_configuration_standard" {
+  template = "${file("${path.module}/templates/cloudwatch_agent_configuration_standard.json")}"
+
+  vars {
+    aggregation_dimensions      = "${jsonencode(var.aggregation_dimensions)}"
+    cpu_resources               = "${var.cpu_resources}"
+    disk_resources              = "${jsonencode(var.disk_resources)}"
+    metrics_collection_interval = "${var.metrics_collection_interval}"
+  }
+}
+
+data "template_cloudinit_config" "cloud_init_merged" {
+  gzip          = true
+  base64_encode = true
+
+  part {
+    filename     = "userdata_part_cloudwatch.cfg"
+    content      = "${data.template_file.cloud_init_cloudwatch_agent.rendered}"
+    content_type = "text/cloud-config"
+  }
+
+  part {
+    filename     = "userdata_part_caller.cfg"
+    content      = "${var.userdata_part_content}"
+    content_type = "${var.userdata_part_content_type}"
+    merge_type   = "${var.userdata_part_merge_type}"
+  }
+}
+
+resource "aws_iam_role" "ec2_cloudwatch" {
+  name = "${module.label.id}"
+
+  assume_role_policy = <<EOF
```

Comment:
let's use `aws_iam_policy_document
https://github.com/cloudposse/terraform-aws-eks-cluster/blob/master/main.tf#L12
instead of HEREDOC

it's our best practice and is easier to read and maintain